### PR TITLE
[v1.12] images/cilium: copy the cilium-envoy binary into Cilium image

### PR DESCRIPTION
**Pull request by aanm**
- [ ] #29340

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 29340; do contrib/backporting/set-labels.py $pr done 1.12; done
```
or with
```
make add-labels BRANCH=v1.14 ISSUES=29340
```

### CODE REVIEW

**Comment by aanm** [comment]
/test-backport-1.12

File: images/cilium/Dockerfile [hunk 1]
```diff
@@ -83,7 +83,8 @@ ARG TARGETOS
 ARG TARGETARCH
 LABEL maintainer="maintainer@cilium.io"
 RUN echo ". /etc/profile.d/bash_completion.sh" >> /etc/bash.bashrc
-COPY --from=cilium-envoy / /
+COPY --from=cilium-envoy /usr/lib/libcilium.so /usr/lib/libcilium.so
```
Comment (sayboras) [inline images/cilium/Dockerfile]: This can be removed as we are still building proxylib (e.g libcilium.so) from cilium/cilium repo in 1.12

**Review by sayboras — APPROVED** [review]

**Comment by sayboras** [comment]
/test-backport-1.12

**Comment by sayboras** [comment]
/test-1.18-4.9

**Comment by sayboras** [comment]
/test-1.20-4.9

**Pull request merged**